### PR TITLE
Theme wam

### PR DESCRIPTION
**Pull request by DannyMurphy**
Hi Guys, here is my first attempt at theming Collective Access. 

Can you guys try it out and let me know what you think.

D

### CODE REVIEW

**Comment by leftclickben** [comment]
I don't know why I replied by email, we should have this discussion on here.  This is the main part of my original email (I'll leave out the part about merging PRs).

---
- The text fields for the Registration Numbers are full width so that they are stacked vertically, but they should be side-by-side on one line.  This is somewhat of a special case so there may need to be specific overrides for regno / idno fields.
- The box on the <del>right</del> left ("Creating New Anthropology Record") could potentially be `position: fixed` and `overflow-y: auto`, because the overflow will hardly ever happen (but is needed for those with tiny screens or who want to try and break things), and it means the menu is there the whole time and you don't have to scroll back to the top to get to a different tab / page of the form.  Alternatively we could use a tiny bit of javascript that goes "if the menu is shorter than the window, use fixed, otherwise use absolute".  I put the `position: fixed` on `#leftNav` and it works but needs a bit of `padding-top` to account for the top nav bar.
- The `border-top` rule on the `.labelInfo.button` class has the effect of separating the "add" button from the things that are being added; I'd make it a `border-bottom` instead, or just remove it.
- Some text areas appear to have an inline `width: inherit` which is overriding the `width: 100%` from the stylesheet.  The inline style can be removed with Firebug and it looks fine, just need to make the page render without it.  An example is "Register Entry" on the Anthropology form.

**Comment by leftclickben** [comment]
Reply from @DannyMurphy 

---

Could you send me a couple of screen shots,
- one of the "Registration Numbers"
- one of "Some text areas appear to have an inline `width: inherit`"

### The box on the <del>right</del> left ("Creating New Anthropology Record")

<del>Do you mean the left hand nav?  If so</del>

I think scrolling with the page is OK. If someone is just browsing
the page they will most likely already have their hand in on the scroll
wheel or if they are editing the form when they save the page it takes
them back to the top of the page. But on long pages it might be good to
add a back to top button like
http://webdesignerwall.com/demo/scroll-to-top/scrolltotop.html

If not can you send me a screen shot.

### `border-top` rule on the `.labelInfo.button`

Will take a look at this.

**Comment by leftclickben** [comment]
Screen shots attached

create-anthro-regno-issue-and-short-textarea-issue-rendered
textarea-inline-style-firebug

**Comment by leftclickben** [comment]
> I think scrolling with the page is OK.....

I know what you're saying but what you end up with when you scroll down is a blank black box on the left hand side, I'm just thinking that area could be more useful if the menu stayed there.  A screen shot won't really show much, but you can try it by putting `position: fixed;` on `#leftNav`, and `padding-top: 50px;` on its first child, `#widgets`.  Then scroll up and down.  (The 50px is a guess, it should be the actual height of the top nav).

**Comment by kehh** [comment]
Wow that certainly is a breath of freshness. 

While the floating sidebar would be nice, once you've created the record you've got the 'record inspector' with details about the record plus a thumbnail if one exists, and you don't want the user to not be able to get to the bottom of the list.

One final point is CA uses textareas for all text fields (don't ask me why), so some look a bit strange with the new form styling. (eg create 'Collecting Event' and then scroll down to 'Sampling Protocol' - Text Value is a bit strange. But I think it's possibly to override the height of the element using the admin so maybe these cases can be worked around using that.

**Comment by leftclickben** [comment]
> While the floating sidebar would be nice, once you've created the record you've got the 'record inspector' with details about the record plus a thumbnail if one exists, 

True, but it also occurs to me that the heading font size and padding could come down significantly.

> and you don't want the user to not be able to get to the bottom of the list.

Well that's what the `overflow-y: auto` is for, but the idea would be that on most people's displays (desktop) it wouldn't need to scroll.  The heading font size and padding, and thumbnail image size, and even the padding in the side nav menu items, could also be responsive so that on smaller displays more space is conserved.

Anyway not a must-have, but I do think it looks a bit strange with the big empty black (dark grey) area off to the side.

**Comment by leftclickben** [comment]
Oh yeah and I missed the part from my original email about how good it looks too :+1:

**Comment by DannyMurphy** [comment]
We might be able to do a quick JS calculation to detect the height of the sidebar menu and the height of the screen. We could then apply a position fixed if screen bigger than menu.

**Comment by DannyMurphy** [comment]
@kehh could you grab a quick screen shot of **'Sampling Protocol' - Text Value is a bit strange**

When Andrew is back tomorrow he has a version with all the data in it so I should be able to look into this and @leftclickben's comment regarding **Registration Numbers**

**Comment by leftclickben** [comment]
Yeah I said that ;-)

> Alternatively we could use a tiny bit of javascript that goes "if the menu is shorter than the window, use fixed, otherwise use absolute"

Except I guess I meant static instead of absolute.

Oh and that would need to happen on window resize as well.  I can write that if you like.

**Comment by leftclickben** [comment]
This is what regno currently looks like (this is from a Crustacea object).  The regno in this case is normally written as "C29566".

regno-current

There are some variation with other object types, e.g. Anthropology also has a third (suffix) field, but the idea is generally the same.

**Comment by leftclickben** [comment]
This is what Anthropology currently looks like:

regno-current-anthro

**Comment by kehh** [comment]
samplingprotocolwierdness
There we go Danny. There are two issues in the screenshot. The first pair of values are on the form when it loads (New > Collecting Event).
The next two pairs of values are added when clicking 'Add sampling protocol'.

**Comment by kehh** [comment]
Also hate to have found this, but one of the admin screens has a section that can't be used with the new theme
Navigate to
Manage > Administration > Metadata Elements > Sampling Protocol and scroll down to 'Sub-elements' - the edit, rearrange and delete buttons all overlap as in the screenshot:
subelementsonadminscreen

And with the default theme:
subelementsdefaulttheme

**Comment by DannyMurphy** [comment]
@kehh I thought we just wanted it to look pretty. You telling me you also want it to work as well? crazy.

**Comment by DannyMurphy** [comment]
Are you doing Reg No with Javascript?

**Comment by leftclickben** [comment]
What do you mean?

**Comment by DannyMurphy** [comment]
Don't worry my version has the reg number in a single field. Andrew has copy and pasted me the inspector code from the main CMIS site which I can use.

**Comment by kehh** [comment]
I see the problem. Different object types have different regno patterns, so if you do New > Cultural Heritage > Anthropology Record you will see different fields from if you do New > Natural History > Arachnology and so on. You're probably safest to just set the height and font of those form elements and maybe make them `display:inline-block;width:auto;` or something similar. Plus there's JS validation that pops in after each form element if there's a problem.

**Comment by DannyMurphy** [comment]
OK I have fixed the big things maybe we can put it on the live site and then see what other errors come up.

File: themes/default/css/_icons.scss [hunk 1]
```diff
@@ -0,0 +1,818 @@
+//_icons.scss
+
+@font-face {
+  font-family: 'Glyphicons Halflings';
+  src: url('http://museum.wa.gov.au/sites/all/themes/wamuseumv3/style/build/fonts/glyphicons-halflings-regular.eot');
```
Comment (kehh) [inline themes/default/css/_icons.scss:5]: Hi Danny,
Are these urls going to be stable?

File: themes/default/css/_main.scss [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
+
+
+
+#main {
+
+	background: #f0f0f0 url(http://museum.wa.gov.au/sites/all/themes/wamuseumv3/style/build/images/backgrounds/escheresque.png) 0 0 repeat;
```
Comment (kehh) [inline themes/default/css/_main.scss:7]: Here's another URL that I just wanted to ensure would be persistent

**Comment by kehh** [comment]
I have merged the PR, with the following items needing work in the future:
- [ ] Should we pull in the non CDN provided resources locally?
- [ ] Dashboard still vanilla aside from going fullwidth
- [ ] Login screen still vanilla
- [ ] `_suckerfish.scss` is completely commented out

**Pull request merged**